### PR TITLE
[BPK-2705] Fix flow types in calendar

### PR DESCRIPTION
**Pull request by k0nserv**
The flow types for the external Calendar component did not match the
internal native components.

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-react-native/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: packages/react-native-bpk-component-calendar/src/common-types.js [hunk 1]
```diff
@@ -34,13 +35,26 @@ export const SELECTION_TYPES = {
 };
 
 export type SelectionType = $Keys<typeof SELECTION_TYPES>;
+export type SelectedDatesChanged = ?(Date[]) => mixed;
+export type NativeEvent = SyntheticEvent<$ReadOnly<{| selectedDates: Date[] |}>,>; // eslint-disable-line prettier/prettier
```
Comment (tiagoengel) [inline packages/react-native-bpk-component-calendar/src/common-types.js]: aren't native dates a number? (timestamp)

Comment (k0nserv) [reply packages/react-native-bpk-component-calendar/src/common-types.js]: Oh yeah you're right

Comment (tiagoengel) [reply packages/react-native-bpk-component-calendar/src/common-types.js]: Also just noticed there is an extra comma at the end.
And you could replace $ReadOnly with `+{| selectedDates: Date[] |}` :)

Comment (k0nserv) [reply packages/react-native-bpk-component-calendar/src/common-types.js]: The extra comma gets added by prettier

Comment (k0nserv) [reply packages/react-native-bpk-component-calendar/src/common-types.js]: I tried replacing it with `+{| selectedDates: number[] |}`, but that causes a syntax error.

**Pull request merged**